### ISSUE TITLE
Question - Is there a parallel or concurrent run support ?

### ISSUE THREAD

**Issue by akash10anand** [thetutlage/japa#29]
## Is there a parallel or concurrent run support ?

**Comment by thetutlage** [comment]
Nope. Plus no plans of adding it too. I want Japa to always run in a single process

**Comment by akash10anand** [comment]
@thetutlage The tests can't run in async manner ?

**Comment by thetutlage** [comment]
Didn't get the question completely. You mean writing tests using async/await?

**Comment by akash10anand** [comment]
Ok, so for example I have test1, test2 and test3, which can run asynchronously. I wanted all three tests to start together without waiting for their previous ones to finish.

**Comment by thetutlage** [comment]
Nope, not yet. All tests are always executed in sequence. Running them parallely should be doable and we can add support for it.

However, it will require the user of the library to write tests carefully, especially the hooks. For example: A hook cleaning the file system, or database before each test will running into race conditions.

**Comment by akash10anand** [comment]
Agreed. May be the hooks can be Awaited by default. Only the tests run inside Promise.all(). Thanks let me see if I can help.
